### ISSUE TITLE
volume_tags on instance (sometimes) clobbers attached EBS volume tags

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#884]
_This issue was originally opened by @gavD as hashicorp/terraform#15302. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version

`0.9.8`

### Description

I use `volume_tags` on an EC2 instance to set my boot volume's tags. However, for my mounted volumes, I want slightly different tags (I use `MakeSnapshot: True` so a custom tool makes a snapshots ONLY of the mounted volume)

### Expected Behavior

MakeSnapshot: True is set on the mounted volume

### Actual Behavior

Terraform SOMETIMES does what I want and SOMETIMES stripes the MakeSnapshot tag. Here's the plan output:

```
~ module.basic2.aws_instance.database.0
    volume_tags.%:            "10" => "9"
    volume_tags.MakeSnapshot: "True" => ""
```

### Steps to Reproduce

Here's an example terraform config with the irrelevant parts stripped out.

```
resource "aws_instance" "database" {

   # ...various config here...

  root_block_device {
    volume_type = "gp2"
  }

  tags {
    Name           =  "myinstance"
  }

  # I only want these on the boot volume
  volume_tags {
    Name           = "myinstance-bootVolume"
  }
}

resource "aws_ebs_volume" "database" {

  # ... various config here...

  type              = "gp2"

  tags {
    Name           = "myinstance-dataVolume"
    MakeSnapshot   = "True"
  }
}

resource "aws_volume_attachment" "database" {
  # ... various config here ...
}
```

then run `terraform plan`

### Important Factoids
- Running in EC2 VPCs
- Sometimes it will strip the tag from the data, sometimes it will not

### References
- GH-14019

**Comment by kevinprince** [comment]
Have encountered same issue with same setup in Terraform 0.9.8 and 0.9.11. 

Attached volume Name tag gets clobbered by the instance volume Name tag.

**Comment by quaeritate** [comment]
This is a very annoying problem.

It could be solved by:

Only allow volume_tags to apply to block devices defined in the aws_instance resource, or shift the tags to the block_device sections.

Or add a new tagging resource that allows tags to be added to the root block device outside of the aws_instance. This ones seems like a really bad idea though.

**Comment by tdmalone** [comment]
Related tickets: #729 & #720 (possibly all duplicates of the same issue?)

**Comment by tdmalone** [comment]
Related tickets: #729 & #770 (possibly all duplicates of the same issue?)

**Comment by YakDriver** [comment]
We have merged a fix to the `volume_tags` issue in #15474. We have added tests to cover the issues observed. Please note that using `volume_tags` in `aws_instance` is _not_ compatible with using `tags` in `aws_ebs_volume`. You need to use one or the other. Prior to this fix, even following this rule, you would encounter errors. Along with the fix, we've added `tags` to the `root_block_device` and `ebs_block_device` configuration blocks in `aws_instance`.

Now that the fix is in place, if you find any problems with `volume_tags`, let us know by [opening a new issue](https://github.com/hashicorp/terraform-provider-aws/issues/new/choose).
